internal/closers.go

```diff
@@ -0,0 +1,17 @@
+package internal
+
+import "io"
+
+type anonymousCloser struct {
+	io.Closer
+	closeFunc func() error
+}
+
+// NewAnonymousCloser is a convenience function to transform anonymous functions into
+// io.Closers
+func NewAnonymousCloser(f func() error) io.Closer {
```

Comment:
putting this in an internal package is really an overkill. you can just put it under root with an unexported type.

also in the current tracing implementation even if user calls `InitGlobalTracer` multiple times with different configs, we always update the same one and it's always the same one registered inside opentracing (so `opentracing.GetGlobalTracer` would always return the same global tracer), so you can just cast that into `*tracing.Tracer` and add it into closers directly.

And that really should be called `closers` instead of `afterStop`.